modules/custom/social_content_export/social_content_export.module

```diff
@@ -0,0 +1,143 @@
+
+/**
+ * @file
+ * The social_content_export module.
+ */
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\views\Views;
+
+/**
+ * Returns event location.
+ *
+ * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+ *   The entity to look for data to export.
+ *
+ * @return mixed
+ *   Return location of event to take place.
+ */
+function social_content_export_event_location(ContentEntityInterface $entity) {
+  $query = \Drupal::database()->select('node__field_event_location', 'nfel');
+  $query->addField('nfel', 'field_event_location_value');
+  $query->condition('nfel.entity_id', $entity->id());
+
+  return $query
+    ->execute()
+    ->fetchField();
+}
+
+/**
+ * Returns event host.
+ *
+ * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+ *   The entity to look for data to export.
+ *
+ * @return mixed
+ *   Returns address of event.
+ */
+function social_content_export_event_host(ContentEntityInterface $entity) {
+  $query = \Drupal::database()->select('node__field_event_address', 'ad');
+  $query->addField('ad', 'field_event_address_locality');
+  $query->condition('ad.entity_id', $entity->id());
+
+  return $query
+    ->execute()
+    ->fetchField();
+}
+
+/**
+ * Returns participant.
+ *
+ * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+ *   The entity to look for data to export.
+ *
+ * @return mixed
+ *   Return event participants.
+ */
+function social_content_export_event_participant(ContentEntityInterface $entity) {
+  $query = \Drupal::database()->select('event_enrollment_field_data', 'efd');
+  $query->addField('efd', 'name');
+  $query->condition('efd.user_id', $entity->id());
+
+  return $query
+    ->execute()
+    ->fetchField();
+}
+
+/**
+ * Returns event title.
+ *
+ * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+ *   The entity to look for data to export.
+ *
+ * @return mixed
+ *   Return event title.
+ */
+function social_content_export_event_title(ContentEntityInterface $entity) {
+  $query = \Drupal::database()->select('node_field_data', 'nfd');
+  $query->addField('nfd', 'title');
+  $query->condition('nfd.nid', $entity->id());
+
+  return $query
+    ->execute()
+    ->fetchField();
+}
+
+/**
+ * Returns event organizer.
+ *
+ * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+ *   The entity to look for data to export.
+ *
+ * @return mixed
+ *   Return event organizer.
+ */
+function social_content_export_event_organizer(ContentEntityInterface $entity) {
+  $query = \Drupal::database()->select('event_enrollment_field_data', 'eo');
+  $query->addField('eo', 'name');
+  $query->condition('eo.id', $entity->id());
+
+  return $query
+    ->execute()
+    ->fetchField();
+}
+
+/**
+ * Implements hook_file_download().
+ */
+function social_content_export_file_download($uri) {
+  $scheme = \Drupal::service('file_system')->uriScheme($uri);
+  $target = file_uri_target($uri);
+  $access = \Drupal::currentUser()->hasPermission('administer node types');
```

Comment:
@bramtenhove or @jaapjan what would be a more appropriate permission to use here? `administer content`?